polars/polars-core/src/chunked_array/kernels/take.rs

```diff
@@ -352,15 +348,13 @@ pub(crate) unsafe fn take_utf8(
 ) -> Arc<LargeStringArray> {
     let data_len = indices.len();
 
-    let offset_len_in_bytes = (data_len + 1) * mem::size_of::<i64>();
-    let mut offset_buf = MutableBuffer::new(offset_len_in_bytes);
-    offset_buf.resize(offset_len_in_bytes, 0);
-    let offset_typed = offset_buf.typed_data_mut();
+    let mut offset_buf = MutableBuffer::<i64>::from_len_zeroed(data_len + 1);
+    let offset_typed = offset_buf.as_mut_slice();
```

Comment:
Note that `typed_data_mut` is basically a transmute and thus `unsafe` (even if not marked as such). This is the equivalent code in arrow2, where we use `as_mut_slice` (or `DeRefMut`) to achieve the same behavior in a safe manner. Note how we also do not need `mem::size_of::<i64>()` to allocate the buffer; the type takes care of that.

Reply:
And safety guaranteed by its type `MutableBuffer<T>`. :)